staging/src/k8s.io/component-base/compatibility/registry_test.go

```diff
@@ -444,6 +446,31 @@ func TestVersionMappingWithCyclicDependency(t *testing.T) {
 	}
 }
 
+func TestAddMetrics(t *testing.T) {
+	r := NewComponentGlobalsRegistry()
+	ver1 := NewEffectiveVersionFromString("0.58", "", "")
+	ver2 := NewEffectiveVersionFromString("1.2", "1.1", "")
+
+	if err := r.Register("comp1", ver1, nil); err != nil {
+		t.Fatalf("expected no error to register new component, but got err: %v", err)
+	}
+	if err := r.Register("comp2", ver2, nil); err != nil {
+		t.Fatalf("expected no error to register new component, but got err: %v", err)
+	}
+	r.AddMetrics()
+
+	expectedOutput := `
+	# HELP version_info [ALPHA] Provides the version info of the component, 
```

Comment:
Would the indentation here cause false negatives

Reply:
I don't think so since the underlying code splits by newline, however good idea to reduce the risk of introducing errors. Removed the newline.

Reply:
got it